src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.Android.cs

```diff
@@ -0,0 +1,89 @@
+        // 2) starting a background thread to load TimeZoneInfo local cache
+        //
+        // On Android, loading AndroidTZData is expensive for startup performance.
+        // The fast result relies on `System.TimeZoneInfo.LocalDateTimeOffset` being set
+        // in the App Context's properties as the appropriate local date time offset from UTC.
+        // monovm_initialize(_preparsed) can be leveraged to do so.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // So, on first call, we return the fast path and start a background thread to load
+        // the TimeZoneInfo Local cache implementation which loads AndroidTZData.
+        public static DateTimeOffset Now
+        {
+            get
+            {
+                DateTime utcDateTime = DateTime.UtcNow;
+
+                if (s_androidTZDataLoaded) // The background thread finished, the cache is loaded.
+                    return ToLocalTime(utcDateTime, true);
+
+                if (s_startNewBackgroundThread) // The cache isn't loaded and no background thread has been created
+                {
+                    lock (s_localUtcOffsetLock)
+                    {
+                        // Now may be called multiple times before a cache is loaded and a background thread is running,
+                        // once the lock is available, check for a cache and background thread.
+                        if (s_androidTZDataLoaded)
+                            return ToLocalTime(utcDateTime, true);
+
+                        if (s_loadAndroidTZData == null)
+                        {
+                            s_loadAndroidTZData = new Thread(() => {
+                                // Delay the background thread to avoid impacting startup, if it still coincides after 1s, startup is already perceived as slow
+                                Thread.Sleep(1000);
+
+                                _ = TimeZoneInfo.Local; // Load AndroidTZData
+                                s_androidTZDataLoaded = true;
+
+                                lock (s_localUtcOffsetLock)
+                                {
+                                    s_loadAndroidTZData = null; // Ensure thread is cleared when cache is loaded
+                                }
+                            });
+                            s_loadAndroidTZData.IsBackground = true;
+                        }
+                    }
+
+                    if (s_startNewBackgroundThread)
+                    {
+                        // Because Start does not block the calling thread,
+                        // setting the boolean flag to false immediately after should
+                        // prevent two calls to DateTimeOffset.Now in quick succession
+                        // from both reaching here.
+                        s_loadAndroidTZData.Start();
+                        s_startNewBackgroundThread = false;
```

Comment:
What prevents the race condition where two threads both read s_startNewBackgroundThread as true? Seems like it'd be safer to have a local that determines whether this thread is the one that gets to start it, and that local is set appropriately inside of the above lock.

Reply:
I believe we made a change based on feedback from @grendello.  I don't recall the specifics as to why.

Reply:
The feedback was about the previous state of the code, where the thread was started within the same lock that the thread then takes, which could end in a deadlock. The assumption here is that since `.Start()` doesn't block the current thread, the flag is set likely before the thread actually is started. Even in the unlikely situation when two threads would attempt to start the worker thread, the only harm would be a little time wasted inside the worker thread. This code could use a semaphore, for instance, but we felt that would be an overkill.

Reply:
> Even in the unlikely situation when two threads would attempt to start the worker thread, the only harm would be a little time wasted inside the worker thread

Calling Start on a thread that's already started will result in an exception.  So the concern I'm raising above is that this could lead to reliability problems, not just performance ones.

Reply:
It's a very unlikely event, but sure, it is better to wrap it in in `try/catch` then and ignore the potential exception. Performance was the main point of this PR, so it would be a shame to sacrifice it to handle a very unlikely case with semaphores or similar mechanisms.